process/smartContract/hooks/blockChainHook.go

```diff
@@ -480,6 +494,42 @@ func (bh *BlockChainHookImpl) SetCurrentHeader(hdr data.HeaderHandler) {
 	bh.mutCurrentHdr.Unlock()
 }
 
+// SaveCompiledCode saves to cache and storage the compiled code
+func (bh *BlockChainHookImpl) SaveCompiledCode(codeHash []byte, code []byte) {
+	bh.compiledScPool.Put(codeHash, code, len(code))
+	err := bh.compiledScStorage.Put(codeHash, code)
+	if err != nil {
+		log.Debug("SaveCompiledCode", "error", err)
+	}
+}
+
+// GetCompiledCode returns the compiled code if it finds in the cache or storage
+func (bh *BlockChainHookImpl) GetCompiledCode(codeHash []byte) (bool, []byte) {
+	val, found := bh.compiledScPool.Get(codeHash)
+	if found {
+		compiledCode, ok := val.([]byte)
+		if ok {
+			return found, compiledCode
+		}
+	}
+
+	compiledCode, err := bh.compiledScStorage.Get(codeHash)
+	if err != nil {
+		return false, nil
+	}
+
+	return found, compiledCode
```

Comment:
return true, compiledCode as "found" could be false here

Reply:
done